### PR TITLE
fix cancel proposal requirement

### PR DESCRIPTION
**Pull request by davidbrai**
it should check that the proposal does not have more votes than the proposal threshold,
i.e less than or equal to proposal threshold should allow canceling

from: https://github.com/code-423n4/2022-08-nounsdao-findings/issues/255

### CODE REVIEW

File: packages/nouns-contracts/test/foundry/NounsDAOLogicV2.t.sol [hunk 1]
```diff
@@ -94,3 +118,55 @@ contract NounsDAOLogicV2Test is Test {
...
+        uint256[] memory values = new uint256[](1);
+        values[0] = 100;
+        string[] memory signatures = new string[](1);
+        signatures[0] = '';
+        bytes[] memory calldatas = new bytes[](1);
+        calldatas[0] = '';
+        proposalId = daoProxy.propose(targets, values, signatures, calldatas, 'my proposal');
+    }
+
+    function testProposerCanCancelProposal() public {
```
[the remark this answers is not in the document]
Comment (eladmallel) [reply packages/nouns-contracts/test/foundry/NounsDAOLogicV2.t.sol:148]: yeah makes sense, i't not a big deal
I'm hoping for a single style of testing
and I do like your separation to test contracts that makes these contracts shorter
so I'm in wait and see mindset
we'll solidify a standard over time

**Comment by davidbrai** [comment]
@solimander please review 🙏

**Comment by solimander** [comment]
Worth fixing for V1 as well?

**Review by solimander — APPROVED** [review]

**Comment by davidbrai** [comment]
@solimander regarding fixing for V1:
my only mental block with that is that the protocol's deployed contracts will differ from the latest code on github.
we'll have the same situation with fixing `delegateBySig`.

do you have any thoughts on that? or it doesn't bother you?

**Pull request merged**